evcouplings/utils/pipeline.py

```diff
@@ -472,6 +476,89 @@ def _handler(signal_, frame):
         raise
 
 
+def calculate_memory_requirements(config):
+    """
+
+    Calculates the memory requirements in MB based on the pipeline configuration
+
+    memory_in_MB = (1/2 * q^2 * (L - 1) * L  + q * L) / 12500
+
+    with:
+        - q: states of Potts model
+        - L: Sequence length
+
+
+    Parameters
+    ----------
+    config : dict
+        A dictionary holding the pipeline configurations
+
+    Returns
+    -------
+    int
+        RAM requirements in MB
+
+    Raises
+    ------
+    ResourceError
+    """
+
+    # gather alignment stages (monomer has only one, complex has 2)
+    stages = filter(lambda x: "align" in x, config["stages"])
+    L = 0
+
+    # extract state information
+    # if alphabet is not explicitly defined assume amino acid alphabet + gap
+    q = len(config["couplings"]["alphabet"]) if config["couplings"]["alphabet"] else 21
+    if config["couplings"]["ignore_gaps"]:
+        q -= 1
+
+    # iterate through alignment stages to determine input sequence length
+    global_cfg = config["global"]
+    for stage in stages:
+        incfg = {
+            **config[stage],
+            **global_cfg
+        }
+
+        # three cases have to be considered
```

Comment:
There is actually 4 cases, so code below is not fully correct - should be checked in that order:
* first check if region is given (standard or existing protocol, starting from sequence or alignment) -> trivial
* if standard protocol: starting from a sequence 
   a) either sequence is given in file (in "sequence_file", sequence_id still needs to be specified for this even though only used to rename the sequence)
   b) only identifier given, need to download
* if existing protocol (and no region): use alignment (in "alignment_file", not in "sequence_file" which in this case is extracted from alignment as the align stage is running); this alignment needs to be focussed or otherwise memory requirements could be vastly over-estimated

The cases for standard protocol (sequence download necessary) can be addressed using the existing fetch_sequence function

Reply:
I didn't want to use fetch_sequence as it would require a temporary output file to store the downloaded sequence.